### PR TITLE
feat: create prototype templates for using uipath with rest connector

### PR DESCRIPTION
**Pull request by mbiciin**
## Description
Add prototype templates to call addQueueItem and getQueueItembyID APIs for UIPath. 

## Related issues
#https://github.com/camunda/product-hub/issues/190

closes #
#135

### CODE REVIEW

File: connectors/uipath/element-templates/uipath-prototype-add-queue-item.json [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+          "value": "oauth-client-credentials-flow"
+        }
+      ],
+      "binding": {
+        "type": "zeebe:input",
+        "name": "authentication.type"
+      }
+    },
+    {
+      "id": "method",
```
Comment (bastiankoerber) [inline connectors/uipath/element-templates/uipath-prototype-add-queue-item.json:85]: please set the needed method and hide it

File: connectors/uipath/element-templates/uipath-prototype-add-queue-item.json [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+        "oneOf": [
+          "post",
+          "put",
+          "patch",
+          "delete"
+        ]
+      }
+    },
+    {
+      "label": "Bearer Token",
```
Comment (bastiankoerber) [inline connectors/uipath/element-templates/uipath-prototype-add-queue-item.json:177]: hide it, if is not needed.

File: connectors/uipath/element-templates/uipath-prototype-add-queue-item.json [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+      "constraints": {
+        "notEmpty": true
+      },
+      "condition": {
+        "property": "authenticationType",
+        "equals": "oauth-client-credentials-flow"
+      }
+    },
+    {
+      "label": "Scopes",
```
Comment (bastiankoerber) [inline connectors/uipath/element-templates/uipath-prototype-add-queue-item.json:248]: can we prefill this with the needed scope and hide it?

File: connectors/uipath/element-templates/uipath-prototype-add-queue-item.json [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+        "notEmpty": true
+      },
+      "condition": {
+        "property": "authenticationType",
+        "equals": "bearer"
+      }
+    },
+    {
+      "label": "OAuth Token Endpoint",
+      "description": "The OAuth token endpoint",
```
Comment (bastiankoerber) [inline connectors/uipath/element-templates/uipath-prototype-add-queue-item.json:195]: prefill it with https://cloud.uipath.com/identity_/connect/token and hide it

File: connectors/uipath/element-templates/uipath-prototype-add-queue-item.json [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+        "type": "zeebe:input",
+        "name": "authentication.scopes"
+      },
+      "condition": {
+        "property": "authenticationType",
+        "equals": "oauth-client-credentials-flow"
+      }
+    },
+    {
+      "label": "Audience",
```
Comment (bastiankoerber) [inline connectors/uipath/element-templates/uipath-prototype-add-queue-item.json:264]: what is it for? Do I need it for this use case?

File: connectors/uipath/element-templates/uipath-prototype-add-queue-item.json [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+      "value": "io.camunda:http-json:1",
+      "binding": {
+        "type": "zeebe:taskDefinition:type"
+      }
+    },
+    {
+      "label": "Type",
+      "id": "authenticationType",
+      "group": "authentication",
+      "description": "Choose the authentication type. Select 'No Auth' if no authentication is necessary",
```
[the remark this answers is not in the document]
Comment (mbiciin) [reply connectors/uipath/element-templates/uipath-prototype-add-queue-item.json]: Hi @bastiankoerber 
As we discussed, I kept just 2 authentication types, oauth and bearer token.

File: connectors/uipath/element-templates/uipath-prototype-add-queue-item.json (hunk 3 above)
Comment (mbiciin) [reply connectors/uipath/element-templates/uipath-prototype-add-queue-item.json:218]: Done!

File: connectors/uipath/element-templates/uipath-prototype-add-queue-item.json (hunk 5 above)
Comment (mbiciin) [reply connectors/uipath/element-templates/uipath-prototype-add-queue-item.json]: I deleted it as it's not required for uipath.

File: connectors/uipath/element-templates/uipath-prototype-add-queue-item.json (hunk 2 above)
Comment (mbiciin) [reply connectors/uipath/element-templates/uipath-prototype-add-queue-item.json:147]: We decided to keep it as it is a valid authentication method for uipath.

**Review by bastiankoerber — CHANGES_REQUESTED** [review]

File: connectors/uipath/element-templates/uipath-prototype-get-item-status.json [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+        }
+      }
+    },
+    {
+      "label": "Filters",
+      "description": "Map of query parameters to add to the request URL",
+      "group": "endpoint",
+      "type": "String",
+      "feel": "required",
+      "value": "{\"$Filter\":\"State eq 'Successful' AND Id eq \"+itemId}",
```
Comment (bastiankoerber) [inline connectors/uipath/element-templates/uipath-prototype-get-item-status.json:109]: why we are filtering for successful here? I think just getting the status is fine: "{"$Filter":"Id eq "+itemId}"

**Review by bastiankoerber — CHANGES_REQUESTED** [review]

File: connectors/uipath/element-templates/uipath-prototype-get-item-status.json [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+        "name": "method"
+      }
+    },
+    {
+      "label": "URL",
+      "description": "Request URL to get the status of the task. Please, replace '{{cloudUrl}}', '{{cloudOrg}}' and '{{cloudTenant}}' with the the values from your uipath orchestrator.",
+      "group": "endpoint",
+      "type": "String",
+      "feel": "required",
+      "value": "=\"https://{{cloudUrl}}/{{cloudOrg}}/{{cloudTenant}}/odata/Jobs",
```
[the remark this answers is not in the document]
Comment (mbiciin) [reply connectors/uipath/element-templates/uipath-prototype-get-item-status.json]: Yes, you're right! I change it!

File: connectors/uipath/element-templates/uipath-prototype-get-item-status.json (hunk 7 above)
Comment (mbiciin) [reply connectors/uipath/element-templates/uipath-prototype-get-item-status.json]: Done! Thank you!

**Pull request closed without merging**